core/quorum_initiator.go

```diff
@@ -656,29 +658,53 @@ func (c *Core) finishConsensus(id string, qt int, p *ipfsport.Peer, status bool,
 	defer c.qlock.Unlock()
 	cs, ok := c.quorumRequest[id]
 	if !ok {
+		fmt.Println("failed to get quorum consensus")
 		if p != nil {
 			p.Close()
 		}
 		return
 	}
+	pd, ok := c.pd[id] //getting details of quorums who pledged
+	if !ok {
+		fmt.Println("failed to get pledged token details")
+		if p != nil {
+			p.Close()
+		}
+		return
+	}
+	keys := make([]string, 0, len(pd.PledgedTokens))
+	for k := range pd.PledgedTokens {
+		keys = append(keys, k)
+	}
+	key := keys[0]
+
 	switch qt {
 	case 0:
 		cs.Result.RunningCount--
 		if status {
 			did := p.GetPeerDID()
-			if cs.Result.SuccessCount < MinConsensusRequired {
-				csig := CreditSignature{
-					Signature:     util.HexToStr(ss),
-					PrivSignature: util.HexToStr(ps),
-					DID:           did,
-					Hash:          hash,
-				}
+			csig := CreditSignature{
```

Comment:
Right, passing the length of quorum list, might work here. So it will be like else if cs.Result.RunsTotal == 6 && cs.Result.SuccessCount == MinConsensusRequired-1

Reply:
I am not sure. Please check.

This is the scenario I am talking about. 

QuorumList : Q1 , Q2 , Q3 , Q4 , Q5, Q6 , Q7

Pledged Quorum : Q1

Initaiate consensus : Q1 , Q2 , Q3 , Q4 respond .   Now say Q5 and Q6 systems are not responding , when you get respond from Q7 , you won't accept it with check `else if cs.Result.RunsTotal == len(quorumList)-1` since RunsTotal is 4 now.